src/Language/Fixpoint/Utils/Profile.hs

```diff
@@ -0,0 +1,13 @@
+module Language.Fixpoint.Utils.Profile where
+
+import Data.Time.Clock
+import Control.Monad
+import Control.Monad.IO.Class
+
+profileIO :: MonadIO m => Bool -> String -> m a -> m a
+profileIO shouldProfile header m = do
+  beforeTime <- liftIO getCurrentTime
+  ret <- m
+  afterTime <- liftIO $ getCurrentTime
+  when shouldProfile $ liftIO $ putStrLn $ header <> show (afterTime `diffUTCTime` beforeTime)
```

Comment:
I think it will work better to collect these measures somewhere and print them at the end of the verification. Also, I imagine that we will want to add up a few of these measures when we call multiple times to the same function.